### ISSUE TITLE
Something went wrong about the result?

### ISSUE THREAD

**Issue by Asteriajojo** [qizhangli/linbp-attack#2]
Hello! I have tried to run your code but the result I got is really confusing T^T
I use test.py to evaluate the result, but only got this 
![image](https://user-images.githubusercontent.com/37069482/111248176-5a8cb100-8644-11eb-83a3-3ba455209061.png)
I use 5000 Imagenet images and their corresponding labels to generate adversarial samples
the parameters I set are like this:
![image](https://user-images.githubusercontent.com/37069482/111248460-e8689c00-8644-11eb-8d41-448614cd1ace.png)
Could you please give me some advice to improve the success rates?
Thanks a lot !!

**Comment by qizhangli** [comment]
Hi!
I've just got the results with these parameters by running
`python3 attack/imagenet/attack_resnet50.py --epsilon 0.1 --niters 300 --batch_size 50 --save_dir github_issue --method linbp_mdi2fgsm`
`python3 attack/imagenet/test.py --dir github_issue`
I got
![image](https://user-images.githubusercontent.com/38599840/111284139-d30e6480-867a-11eb-96a0-85cff342876b.png)


<details>
<summary>Also, here is the result of the `pip list` from the environment I used</summary>

```
Package                Version
---------------------- --------
cycler                 0.10.0
distro-info            0.21
iotop                  0.6
joblib                 1.0.0
kiwisolver             1.3.1
matplotlib             3.3.3
numpy                  1.19.5
opencv-python-headless 4.5.1.48
pandas                 1.2.1
patsy                  0.5.1
Pillow                 8.1.0
pip                    20.3.1
protobuf               3.14.0
pycurl                 7.43.0.2
PyGObject              3.30.4
pyparsing              2.4.7
python-apt             1.8.4.2
python-dateutil        2.8.1
pytz                   2020.5
scikit-learn           0.24.1
scipy                  1.6.0
setuptools             42.0.2
six                    1.15.0
sklearn                0.0
statsmodels            0.12.1
tfrecord               1.11
threadpoolctl          2.1.0
torch                  1.7.1
torchvision            0.8.2
typing-extensions      3.7.4.3
unattended-upgrades    0.1
wheel                  0.36.1

**Comment by qizhangli** [comment]
Could you provide your `selected_imagenet.csv` file?

**Comment by Asteriajojo** [comment]
Hi! I checked my csv.file. 
My label index is from 0 to 999,but if I change it from 1 to 1000, or I use 
![image](https://user-images.githubusercontent.com/37069482/111650825-a2782780-8840-11eb-9b14-2d83092f0e75.png) then the success rate will be 
![image](https://user-images.githubusercontent.com/37069482/111650911-ba4fab80-8840-11eb-8bc9-9ddbea190837.png).
It seems the label mismatching caused my problem. But I wonder if this label mismatching problem will cause some trouble during the attack process?

**Comment by Asteriajojo** [comment]
Hi! I checked my csv.file.
![image](https://user-images.githubusercontent.com/37069482/111654927-2849a200-8844-11eb-9a90-35d44d9b6ac6.png)
My label index is from 0 to 999,but if I change it from 1 to 1000, or I use 
![image](https://user-images.githubusercontent.com/37069482/111650825-a2782780-8840-11eb-9b14-2d83092f0e75.png) then the success rate will be 
![image](https://user-images.githubusercontent.com/37069482/111650911-ba4fab80-8840-11eb-8bc9-9ddbea190837.png).
It seems the label mismatching caused my problem. But I wonder if this label mismatching problem will cause some trouble during the attack process?
Thanks a lot!
